docs/section22/Rule22-25.md

```diff
@@ -0,0 +1,52 @@
+
+# CHW&CW - Rule 22-25  
+
+**Schema Version:** 0.0.10 **Mandatory Rule:** True
+**Rule ID:** 22-25  
+**Rule Description:** Chilled-water systems served by chiller(s) shall be modeled as primary/secondary systems with constant flow primary loop.  
+**Rule Assertion:** B-RMR = expected value  
+**Appendix G Section:** Section 22 CHW&CW Loop  
+**90.1 Section Reference:** Section G3.1.3.10 Chilled-water pumps (Systems 7, 8, 11, 12, and 13)  
+**Data Lookup:** None  
+**Evaluation Context:** Building  
+**Applicability Checks:**  
+
+1. B-RMR is modeled with at least one air-side system that is Type-7, 8, 11.1, 11.2, 12, 13, 7b, 8b, 11b, 12b, 13b.
+2. Primary and secondary CHW loops are modeled correctly in B-RMR.
+
+**Function Calls:**  
+
+1. get_baseline_system_types()
+2. get_primary_secondary_loops()
+
+**Applicability Checks:**  
+
+- **Check 1:** Get B-RMR system types: `baseline_hvac_system_dict = get_baseline_system_types(B-RMR)`
+
+  - Check if B-RMR is modeled with at least one air-side system that is Type-7, 8, 11.1, 11.2, 12, 13, 7b, 8b, 11b, 12b, 13b, i.e. with air-side system served by chiller(s), continue to the next applicability check: `if any(sys_type in baseline_hvac_system_dict.keys() for sys_type in ["SYS-7", "SYS-8", "SYS-11.1", "SYS-11.2", "SYS-12", "SYS-13", "SYS-7B", "SYS-8B", "SYS-11B", "SYS-12B", "SYS-13B"]): NEXT_APPLICABILITY_CHECK`
+
+  - Else, rule is not applicable to B-RMR: `else: RULE_NOT_APPLICABLE`
+
+- **Check 2:** Get primary and secondary loops for B-RMR: `primary_secondary_loop_dictionary = get_primary_secondary_loops(B_RMR, "COOLING")`
+
+  - Check if primary and secondary loops are modeled correctly, continue to rule logic: `if primary_secondary_loop_dictionary["PRIMARY"] AND primary_secondary_loop_dictionary["SECONDARY"]: CHECK_RULE_LOGIC`
+
+  - Else, rule is not applicable to B-RMR: `else: RULE_NOT_APPLICABLE`
+
+## Rule Logic:
+
+- For each pump in B_RMR: `for pump_b in B_RMR.RulesetModelInstance.pumps:`
+
+  - Save loop served by pump and pump to a dictionary: `loop_pump_dictionary[pump_b.loop_or_piping].append(pump)`
+
+- For each primary CHW loop: `for loop_b_id in primary_secondary_loop_dictionary["PRIMARY"]:`
+
+  - For each pump serving the loop: `for pump_b in loop_pump_dictionary[loop_id_b]:`
+
+    **Rule Assertion - Component:**
+
+    - Case 1: If pump serving primary loop is modeled as constant flow: `if pump_b.speed_control == FIXED_SPEED: PASS`
+
```

Comment:
This is now checked in RDS 22-36.

Reply:
@mkarpman slightly confused : does 22-36 need to be a separate rule or should 22-25 confirm all 3 for the primary pump

FluidLoop.cooling_or_condensing_design_and_control.operation="continuous" and
FluidLoop.cooling_or_condensing_design_and_control.flow_control="fixed_flow"
pump = 'Fixed Speed'

Reply:
It doesn't have to be a separate RDS as long as all three are checked.

Reply:
needs to be updates per MK's comment in 22-36 #368

Reply:
Needs to be updated per comemnt in #368-  
I would consider splitting this into two rules - (1) 5-25 : continuous constant flow primary and (2) 5-36: continuous variable flow secondary. This would allow for a more targeted error message (i.e., better understanding of what exactly was wrong).

Reply:
discussed on 04.26.2022 - will keep it the way it is. 22-25 will check constant flow and 22-36 will check continuos + secondary variable flow.

Reply:
As discussed,
Pump rules:
22-25 would verify the primary pump is constant speed.
22-10 would verify the secondary pump is constant speed (<300ton) or with VFD (>=300ton).
Loop rules:
22-7 would verify that baseline has been modeled with P S configuration.
22-36 would verify continuous constant flow primary loop and variable-flow secondary loop.
22-34 would verify baseline has been modeled with only one CHW plant (i.e. one primary and one secondary loop).